### PR TITLE
Instance Refactor

### PR DESCRIPTION
**Pull request by CADIndie**
Refactors majority of instance management to match Pojlibs changes/refactor: QuestCraftPlusPlus/Pojlib#39

* Fixes ModManager Mod Downloading
* Adds Data Packs
* Adds Resource Packs
* Adds Mod Packs
* Adds Custom Instance Creation
* Redoes Some UI Elements
* Plus More!

### CODE REVIEW

File: Assets/Scripts/TextScroll.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using TMPro;
+using UnityEngine;
+
+public class TextScroll : MonoBehaviour
+{
+    public float scrollSpeed = 50f;
+    private RectTransform rectTransform;
+    private TextMeshProUGUI textComponent;
+
+    /*private void Update()
```
Comment (thejudge156) [inline Assets/Scripts/TextScroll.cs:13]: Is this still needed?

**Comment by thejudge156** [comment]
Other than 1 little question, looks good.

File: Assets/Scripts/TextScroll.cs (hunk 1 above)
Comment (thejudge156) [reply Assets/Scripts/TextScroll.cs:13]: Talked about in VC, waiting for commit to remove it

**Review by thejudge156 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/TextScroll.cs (hunk 1 above)
Comment (CADIndie) [reply Assets/Scripts/TextScroll.cs]: Removed the script as its useless at the moment, will possibly add in UX PR.

**Review by thejudge156 — APPROVED** [review]

**Pull request merged**